File: src/main/java/pl/mroz/buddiesapi/infrastructure/database/account/AccountRepositoryConfig.java
package pl.mroz.buddiesapi.infrastructure.database.account;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import pl.mroz.buddiesapi.domain.account.AccountRepository;

@Configuration
public class AccountRepositoryConfig {

    @Bean
    AccountRepository accountRepository(AccountRepositoryImpl.AccountJpaRepository jpaRepository,
                                        AccountRepositoryImpl.LocationJpaRepository locationJpaRepository) {
        return new AccountRepositoryImpl(jpaRepository, locationJpaRepository);
    }
}

File: src/main/kotlin/buddiesapi/infrastructure/database/account/AccountRepositoryConfig.kt
package buddiesapi.infrastructure.database.account

import buddiesapi.domain.account.AccountRepository
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration


@Configuration
class AccountRepositoryConfig {
    @Bean
    fun accountRepository(jpaRepository: AccountRepositoryImpl.AccountJpaRepository): AccountRepository {
        return AccountRepositoryImpl(jpaRepository)
    }
}
